doc/source/dev/extend.rst

```diff
@@ -6,17 +6,17 @@ ODL is written to be easy to extend with new functionality and classes, and new
 
 There are several ways to extend ODL, some of which are listed below.
 
-Adding Fn spaces
-----------------
-The abstract spaces `FnBase` and `NtuplesBase` are the workhorses of the ODL space machinery. They are used in both the discrete :math:`R^n` case, as well as data representation for discretized function spaces such as :math:`L^2([0, 1])` in the `DiscretizedSpace` class. These are in general created through the `rn` and `uniform_discr` functions who take an ``impl`` parameter, allowing users to select the backend to use.
+Adding tensor spaces
+--------------------
+The abstract spaces `BaseTensorSpace` and `BaseTensorSet` are the workhorses of the ODL space machinery. They are used in both the discrete :math:`R^n` case, as well as data representation for discretized function spaces such as :math:`L^2([0, 1])` in the `DiscretizedSpace` class. These are in general created through the `rn` and `uniform_discr` functions who take an ``impl`` parameter, allowing users to select the backend to use.
```

Comment:
I'd only make the obvious changes like switching names and adapting headings, perhaps replace obsolete passages. More in-depth work on docs later.